modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/ExceptionUtils.cs

```diff
@@ -135,5 +135,122 @@ public static void LogUnhandledException(Exception e)
+        {
+            string methodstr = method;
+            string basestr = GetVariantTypeName(instance);
+
+            switch (method)
+            {
+                case "Call":
+                    if (argCount >= 1 && instance->Type != Variant.Type.Callable)
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*args[0])} (via call)";
+                    }
+                    else
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*instance)} (Callable)";
+                    }
+                    break;
+                case "CallRecursive" when basestr == "Godot.TreeItem":
+                    if (argCount >= 1)
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*args[0])} (via TreeItem.CallRecursive)";
+                    }
+                    break;
+            }
+
+            methodstr = $"function '{methodstr}'";
+            if (instance->Type != Variant.Type.Callable)
+            {
+                methodstr += $" in base '{basestr}'";
+            }
+
+            return methodstr;
+        }
+
+        private unsafe static string GetCallErrorMessage(godot_variant_call_error error, string where, godot_variant** args)
+        {
+            switch (error.Error)
+            {
+                case godot_variant_call_error_error.GODOT_CALL_ERROR_CALL_ERROR_INVALID_ARGUMENT:
+                {
+                    int errorarg = error.Argument;
+                    // Handle the Object to Object case separately as we don't have further class details.
+#if DEBUG
+                    if (error.Expected == Variant.Type.Object && args[errorarg]->Type == error.Expected)
+                    {
+                        return $"Invalid type in {where}. The Object-derived class of argument {errorarg + 1} (" + GetVariantTypeName(args[errorarg]) + ") is not a subclass of the expected argument class.";
+                    }
+                    else if (error.Expected == Variant.Type.Array && args[errorarg]->Type == error.Expected)
+                    {
+                        return $"Invalid type in {where}. The array of argument {errorarg + 1} (" + GetVariantTypeName(args[errorarg]) + ") does not have the same element type as the expected typed array argument.";
+                    }
+                    else
+#endif
+                    {
+                        return $"Invalid type in {where}. Cannot convert argument {errorarg + 1} from {args[errorarg]->Type} to {error.Expected}.";
+                    }
+                }
+                case godot_variant_call_error_error.GODOT_CALL_ERROR_CALL_ERROR_TOO_MANY_ARGUMENTS:
+                case godot_variant_call_error_error.GODOT_CALL_ERROR_CALL_ERROR_TOO_FEW_ARGUMENTS:
+                    return $"Invalid call to {where}. Expected {error.Argument} arguments.";
```

Comment:
This case doesn't seem to get triggered and instead the invalid method case is used.

Reply:
When the method can't be found, we always use `CALL_ERROR_INVALID_METHOD` regardless of the arg count. See https://github.com/godotengine/godot/issues/69519.

https://github.com/godotengine/godot/blob/f438fdb752cc3417d3926c20c50c92d615cfca0c/modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/CSharpInstanceBridge.cs#L24-L34

I added all the enum cases assuming we may change this behavior in the future and decide to use the other ones.

Reply:
By the way, I just realized the other enums will be used by GDScript, so for example:

```csharp
var gdnode = GetNode("GDScriptNode");
gdnode.Call("my_gdscript_method");
```